pkg/controller/node/rate_limited_queue.go

```diff
@@ -216,3 +219,24 @@ func (q *RateLimitedTimedQueue) Remove(value string) bool {
 func (q *RateLimitedTimedQueue) Clear() {
 	q.queue.Clear()
 }
+
+// SwapLimiter safely swaps current limiter for this queue with the passed one if capacities or qps's differ.
+func (q *RateLimitedTimedQueue) SwapLimiter(limiter flowcontrol.RateLimiter) {
+	if q.limiter.Capacity() == limiter.Capacity() && q.limiter.QPS() == limiter.QPS() {
+		return
+	}
+	q.limiterLock.Lock()
+	defer q.limiterLock.Unlock()
+	// If we're currently waiting on limiter, we drain the new one - this is best effort approach.
+	for q.limiter.Saturation() > limiter.Saturation() {
```

Comment:
I think we should so it differently.
Let me analyze it on the example. Let's say that both rate-limitters operate on the same period of time and:
1. the old allows for 100 operations, the new 10
2. the old allows for 10 operations, the new for 100

Ad 1. If the old one is saturate in 30%, that means, that we already used 30, so we shouldn't allow for more right now.
Ad 2. If the old is fully saturate, the we still should allow for 90 more immediately.

That said, I think that the correct semantic would be to reduce the new limitter by:
min(old_limiter.Capacity() * old_limiter.Saturation(), new_limiter.Capacity())

But this is really a semantic decision.
@davidopp - What do you think about it?